include/copc-lib/las/point.hpp

```diff
@@ -1,597 +1,590 @@
 #ifndef COPCLIB_LAS_POINT_H_
 #define COPCLIB_LAS_POINT_H_
 
-#include <cstdint>
-#include <iostream>
 #include <sstream>
-#include <stdexcept>
 #include <vector>
 
-#include <copc-lib/las/internal/extra_bytes.hpp>
-#include <copc-lib/las/internal/gps_time.hpp>
-#include <copc-lib/las/internal/nir.hpp>
-#include <copc-lib/las/internal/point_10.hpp>
-#include <copc-lib/las/internal/point_14.hpp>
-#include <copc-lib/las/internal/rgb.hpp>
-#include <copc-lib/las/internal/utils.hpp>
-
 namespace copc::las
 {
 
 class Point
 {
   public:
-    Point(const uint8_t &point_format_id)
-    {
-        point_10_ = nullptr;
-        point_14_ = nullptr;
-        rgb_ = nullptr;
-        gps_time_ = nullptr;
-        nir_ = nullptr;
+    Point() = default;
```

Comment:
let's make this constructor private and add a public constructor that takes the point format and point size as args

Reply:
Currently I get that information while unpacking, you're suggesting we would provide that info at construction and then use it when unpacking?

Reply:
Yeah, something like that. So the user can create points of a certain point type manually. Something like this:

```
static Point Unpack(in_stream, point_id, point_length)
{
	Point point(point_id, point_length);
	x_ = ...
}
```

To enable Unpack to be able to set the private fields of the point, you can mark it as a friend function (see here for static functions https://stackoverflow.com/a/19845672).